cli/ops/testing.rs

```diff
@@ -25,28 +24,32 @@ use std::sync::atomic::AtomicUsize;
 use std::sync::atomic::Ordering;
 use uuid::Uuid;
 
-pub fn init(
-  sender: TestEventSender,
-  fail_fast_tracker: FailFastTracker,
-  filter: TestFilter,
-) -> Extension {
-  Extension::builder("deno_test")
-    .ops(vec![
-      op_pledge_test_permissions::decl(),
-      op_restore_test_permissions::decl(),
-      op_get_test_origin::decl(),
-      op_register_test::decl(),
-      op_register_test_step::decl(),
-      op_dispatch_test_event::decl(),
-      op_tests_should_stop::decl(),
-    ])
-    .state(move |state| {
-      state.put(sender.clone());
-      state.put(fail_fast_tracker.clone());
-      state.put(filter.clone());
-    })
-    .build()
-}
+deno_core::ops!(
+  deno_ops,
+  [
+    op_pledge_test_permissions,
+    op_restore_test_permissions,
+    op_get_test_origin,
+    op_register_test,
+    op_register_test_step,
+    op_dispatch_test_event,
+    op_tests_should_stop,
+  ]
+);
+
+deno_core::extension!(deno_test,
+  ops = deno_ops,
+  config = {
+    sender: TestEventSender,
+    fail_fast_tracker: FailFastTracker,
+    filter: TestFilter,
+  },
+  state = |state, sender, fail_fast_tracker, filter| {
```

Comment:
@littledivy suggested to me that maybe we should have a single struct instead of `config` and then `state` fn would always receive either 1 (`state`) or 2 (`state` and config struct) arguments. Any thought @mmastrac?

Reply:
That would work as well. I actually had that in an earlier pass, but I was worried that having to prefix everything w/`cfg.` added a lot of noise. I'll circle back at the end on this one as well.